apps/namadillo/src/atoms/balance/atoms.ts

```diff
@@ -26,13 +28,35 @@ export type TokenBalance = AddressWithAsset & {
   dollar?: BigNumber;
 };
 
+const DEPRECATED_getViewingKey = async (): Promise<string | undefined> => {
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const namada: Namada | undefined = (window as any).namada;
+    if (namada && semver.lt(namada.version(), "0.3.3")) {
+      const accounts = await namada.accounts();
+      const defaultAccount = await namada.defaultAccount();
+      const shieldedAccount = accounts?.find(
+        (a) => a.type === "shielded-keys" && a.alias === defaultAccount?.alias
+      );
+      return shieldedAccount?.owner;
+    }
+  } catch {
+    // do nothing
+  }
+  return undefined;
+};
+
 export const viewingKeyAtom = atomWithQuery<string>((get) => {
   const accountsQuery = get(accountsAtom);
   const defaultAccountQuery = get(defaultAccountAtom);
 
   return {
     queryKey: ["viewing-key", accountsQuery.data, defaultAccountQuery.data],
     ...queryDependentFn(async () => {
+      const deprecatedViewingKey = await DEPRECATED_getViewingKey();
+      if (deprecatedViewingKey) {
+        return deprecatedViewingKey;
+      }
       const shieldedAccount = accountsQuery.data?.find(
         (a) => a.isShielded && a.alias === defaultAccountQuery.data?.alias
```

Comment:
I know it's not a part of your PR, but the extension `v0.3.3` will also remove that `isShielded`, so maybe we want to get ahead of it and switch `a.isShielded` to `a.type === AccountType.ShieldedKeys` (if you want! Otherwise I'll rebase to this and fix in that PR, but the `a.type` version will work with 0.3.2 too!)